File: GP_Cpp/RuleSet.cpp
#include "RuleSet.h"


/*********************************************************************
* Comment
*********************************************************************/
RuleSet::RuleSet(string symbol)
{
   this->symbol = symbol;
}

/*********************************************************************
* Comment
*********************************************************************/
RuleSet::~RuleSet() {}

string RuleSet::getSymbol() {
   return symbol;
}

/*********************************************************************
* Comment
*********************************************************************/
void RuleSet::addTerminalRule(Rule r)
{
   Terminals.push_back(r);
}

void RuleSet::addNonTerminalRule(Rule r)
{
   NonTerminals.push_back(r);
}

/*********************************************************************
* Comment
*********************************************************************/
Rule RuleSet::getTerminalRule(int i)
{
   return Terminals.at(i);
}

Rule RuleSet::getNonTerminalRule(int i)
{
   return NonTerminals.at(i);
}

/*********************************************************************
* Comment
*********************************************************************/
int RuleSet::numTerminals()
{
   return Terminals.size();
}
/*********************************************************************
* Comment
*********************************************************************/
int RuleSet::numNonTerminals()
{
   return NonTerminals.size();
}

/*********************************************************************
* Comment
*********************************************************************/
bool RuleSet::onlyTerminals()
{
   return (Terminals.size() > 0 && NonTerminals.size() == 0);
}

/*********************************************************************
* Comment
*********************************************************************/
bool RuleSet::onlyNonTerminals()
{
   return (NonTerminals.size() > 0 && Terminals.size() == 0);
}

File: GP_Python/ruleSet.py
from rule import Rule

class RuleSet:
    def __init__(self,symbol):
        self.symbol = symbol
        self.Terminals = []
        self.NonTerminals = []

    def addTerminalRule(self, rule):
        self.Terminals.append(rule)

    def addNonTerminalRule(self, rule):
        self.NonTerminals.append(rule)
    
    def numTerminals(self) -> int:
        return len(self.Terminals)
    
    def numNonTerminals(self) -> int:
        return len(self.NonTerminals)

    def onlyTerminals(self) -> bool:
        return (self.numTerminals()>0 and self.numNonTerminals()==0)

    def onlyNonTerminals(self) -> bool:
        return (self.numNonTerminals()>0 and self.numTerminals()==0)

    